### PR TITLE
[70] Implement Admin's job page

### PR DESCRIPTION
**Pull request by priasdevo**
## What did you do

- image
- Move job-list page to components add new components (Admin page)
- refactor job/list structure
- add chips/grey out to jobCard
- connect with Job/reports

## Demo

- https://dev.modela.miello.dev/job ( use admin account )

### CODE REVIEW

File: apps/web/src/modules/job/list/components/ActorCastingPage/hooks/useJobListData/index.ts [hunk 1]
```diff
@@ -5,7 +5,12 @@ import { apiClient } from 'common/utils/api'
 import dayjs from 'dayjs'
 import { useCallback, useEffect, useRef, useState } from 'react'
 
-import { IFilter, initialIFilter, initialISearch, ISearch } from '../../types'
+import {
+  IFilter,
+  initialIFilter,
+  initialISearch,
+  ISearch,
+} from '../../../../pages/types'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/ActorCastingPage/hooks/useJobListData/index.ts:13]:
```suggestion
} from 'modules/job/list/pages/types'
```

File: apps/web/src/modules/job/list/components/ActorCastingPage/hooks/useJobListData/index.ts [hunk 2]
```diff
@@ -100,7 +105,7 @@ const useJobListData = () => {
   }, [page, search])
 
   useEffect(() => {
-    if (!isOpen) {
+    if (!isOpen || window.innerWidth > 900) {
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/ActorCastingPage/hooks/useJobListData/index.ts:108]: useMediaQuery?

File: apps/web/src/modules/job/list/components/AdminPage/hooks/useJobListData/index.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+import { apiClient } from 'common/utils/api'
+import dayjs from 'dayjs'
+import { useCallback, useEffect, useRef, useState } from 'react'
+
+import {
+  IFilter,
+  initialIFilter,
+  initialISearchAdmin,
+  ISearch,
+} from '../../../../pages/types'
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/AdminPage/hooks/useJobListData/index.ts:13]:
```suggestion
} from 'modules/job/list/pages/types'
```

File: apps/web/src/modules/job/list/components/AdminPage/index.tsx [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    <div
+      style={{
+        display: 'flex',
+        flexDirection: 'row',
+        justifyContent: 'center',
+        alignItems: 'flex-start',
+        gap: '3.5vw',
+      }}
+    >
+      <PlaceFill></PlaceFill>
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/AdminPage/index.tsx:50]:
```suggestion
      <PlaceFill/>
```

File: apps/web/src/modules/job/list/components/AdminPage/index.tsx [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                width: '100%',
+                maxWidth: '250px',
+                marginLeft: '1rem',
+                fontSize: '16px',
+              }}
+              startIcon={<Add />}
+            >
+              สร้างงาน
+            </Button>
+          )}
```
Comment (PKhing) [inline apps/web/src/modules/job/list/components/AdminPage/index.tsx:116]: remove this since casting is unable to access this page

**Review by PKhing — CHANGES_REQUESTED** [review]
Both AdminPage and ActorCastingPage should be in the pages folder (since it is a page 😂 )

**Comment by PKhing** [comment]
check or uncheck "งานที่ได้รับการแจ้งปัญหา" has no different now. I think this might be from backend since the request is correct (or is it expected behavior @SuitedIceZ )

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 6]
```diff
@@ -34,6 +34,42 @@ export default function FilterContainer(props: FilterContainerProps) {
...
+                งานที่ได้รับการแจ้งปัญหา
+              </Typography>
+            }
+          />
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={state.cancleCheck}
+                onChange={(event) =>
+                  setState({ ...state, cancleCheck: event.target.checked })
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:64]:
```suggestion
                  setState((prev) => ({ ...prev, cancleCheck: event.target.checked }))
```

File: apps/web/src/modules/job/list/pages/types.ts [hunk 7]
```diff
@@ -15,6 +15,8 @@ export interface IFilter {
   maleCheck: boolean
   femaleCheck: boolean
   otherCheck: boolean
+  cancleCheck: boolean
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/types.ts:18]: Please change all object key and types which related to this also
```suggestion
  cancelCheck: boolean
```

File: apps/web/src/modules/job/list/pages/types.ts [hunk 8]
```diff
@@ -47,6 +49,8 @@ export const initialIFilter: IFilter = {
   maleCheck: false,
   femaleCheck: false,
   otherCheck: false,
+  cancleCheck: false,
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/types.ts:52]: Please change all object key and types which related to this also
```suggestion
  cancelCheck: false,
```

File: apps/web/src/modules/job/list/components/FilterContainer/index.tsx [hunk 9]
```diff
@@ -34,6 +34,42 @@ export default function FilterContainer(props: FilterContainerProps) {
...
+        >
+          <Typography variant="body1" sx={{ justifyContent: 'left' }}>
+            แจ้งปัญหา
+          </Typography>
+          <FormControlLabel
+            control={
+              <Checkbox
+                checked={state.reportCheck}
+                onChange={(event) =>
+                  setState({ ...state, reportCheck: event.target.checked })
```
Comment (miello) [inline apps/web/src/modules/job/list/components/FilterContainer/index.tsx:49]:
```suggestion
                  setState((prev) => ({ ...prev, reportCheck: event.target.checked }))
```

File: apps/web/src/modules/job/list/components/AdminPage/hooks/useJobListData/index.ts [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+        setHasMore(res.maxPage > page)
+      }
+      setPage((prev) => prev + 1)
+    } catch (err) {
+      handleError(err)
+    }
+  }, [page, search])
+
+  useEffect(() => {
+    console.log(window.innerWidth)
```
Comment (miello) [inline apps/web/src/modules/job/list/components/AdminPage/hooks/useJobListData/index.ts:115]: Consider removing `console.log`

**Review by miello — CHANGES_REQUESTED** [review]

File: apps/web/src/modules/job/list/pages/types.ts (hunk 7 above)
Comment (PKhing) [reply apps/web/src/modules/job/list/pages/types.ts:18]: ตาโคตรดี :joy:

**Comment by priasdevo** [comment]
> check or uncheck "งานที่ได้รับการแจ้งปัญหา" has no different now. I think this might be from backend since the request is correct (or is it expected behavior @SuitedIceZ )

เหมือนว่า database ของ dev อันที่ open อยู่ทุกอันโดน report ( เปิด close ด้วยหรือเช็คกับ beta ไม่มีปัญหา)

File: apps/web/src/modules/job/list/pages/ActorCastingPage/index.tsx [hunk 11]
```diff
@@ -0,0 +1,157 @@
+import { UserType } from '@modela/database'
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/ActorCastingPage/index.tsx:1]:
```suggestion
import { UserType } from '@modela/dtos'
```

File: apps/web/src/modules/job/list/pages/index.tsx [hunk 12]
```diff
@@ -1,152 +1,18 @@
 import { UserType } from '@modela/database'
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/index.tsx:1]:
```suggestion
import { UserType } from '@modela/dtos'
```

File: apps/web/src/modules/job/list/pages/AdminPage/index.tsx [hunk 13]
```diff
@@ -0,0 +1,122 @@
+import { UserType } from '@modela/database'
```
Comment (miello) [inline apps/web/src/modules/job/list/pages/AdminPage/index.tsx:1]:
```suggestion
import { UserType } from '@modela/dtos'
```

**Review by miello — CHANGES_REQUESTED** [review]
That's all from me

**Review by miello — APPROVED** [review]
Please wait for review from @PKhing also

**Review by PKhing — APPROVED** [review]

**Pull request merged**